runtime/compiler/aarch64/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -36,8 +40,1040 @@ void J9::ARM64::AheadOfTimeCompile::processRelocations()
+            uint16_t receiverClassID = symValManager->getIDFromSymbol(static_cast<void *>(thisClass));
+
+            uintptrj_t data = 0;
+            data = (((uintptrj_t)receiverClassID << 16) | (uintptrj_t)methodID);
+            *(uintptrj_t*)cursor = data;
+            }
+         else
+            {
+            *(uintptrj_t*)cursor = (uintptrj_t)guard->getSymbolReference()->getCPIndex(); // record cpIndex
+            }
+         cursor += SIZEPOINTER;
+
+         TR_OpaqueClassBlock *inlinedMethodClass = resolvedMethod->containingClass();
+         void *romClass = (void *) fej9->getPersistentClassPointerFromClassPointer(inlinedMethodClass);
+         uintptr_t romClassOffsetInSharedCache = sharedCache->offsetInSharedCacheFromPointer(romClass);
+
+         *(uintptrj_t *) cursor = romClassOffsetInSharedCache;
+         cursor += SIZEPOINTER;
+
+         if (relocation->getTargetKind() != TR_InlinedInterfaceMethod &&
+             relocation->getTargetKind() != TR_InlinedVirtualMethod)
+            {
+            *(uintptrj_t *) cursor = (uintptrj_t) relocation->getTargetAddress(); // record Patch Destination Address
+            cursor += SIZEPOINTER;
+            }
+         }
+         break;
+
+      case TR_ProfiledInlinedMethodRelocation:
+      case TR_ProfiledClassGuardRelocation:
+      case TR_ProfiledMethodGuardRelocation:
+         {
+         guard = (TR_VirtualGuard *) relocation->getTargetAddress2();
+
+         int32_t inlinedSiteIndex = guard->getCurrentInlinedSiteIndex();
+         *(uintptrj_t *) cursor = (uintptrj_t) inlinedSiteIndex;
+         cursor += SIZEPOINTER;
+
+         TR::SymbolReference *callSymRef = guard->getSymbolReference();
+         TR_ResolvedMethod *owningMethod = callSymRef->getOwningMethod(comp);
+
+         TR_InlinedCallSite & ics = comp->getInlinedCallSite(inlinedSiteIndex);
+         TR_ResolvedMethod *inlinedMethod = ((TR_AOTMethodInfo *)ics._methodInfo)->resolvedMethod;
+         TR_OpaqueClassBlock *inlinedCodeClass = reinterpret_cast<TR_OpaqueClassBlock *>(inlinedMethod->classOfMethod());
+
+         *(uintptrj_t *) cursor = (uintptrj_t) owningMethod->constantPool(); // record constant pool
+         cursor += SIZEPOINTER;
+
+         if (comp->getOption(TR_UseSymbolValidationManager))
+            {
+            uint16_t inlinedCodeClassID = symValManager->getIDFromSymbol(static_cast<void *>(inlinedCodeClass));
+            uintptrj_t data = (uintptrj_t)inlinedCodeClassID;
+            *(uintptrj_t*)cursor = data;
+            }
+         else
+            {
+            *(uintptrj_t*)cursor = (uintptrj_t)callSymRef->getCPIndex(); // record cpIndex
+            }
+         cursor += SIZEPOINTER;
+
+         void *romClass = (void *) fej9->getPersistentClassPointerFromClassPointer(inlinedCodeClass);
+         uintptr_t romClassOffsetInSharedCache = sharedCache->offsetInSharedCacheFromPointer(romClass);
```

Comment:
If https://github.com/eclipse/openj9/pull/7379 gets merged first, this will need to change to `self()->offsetInSharedCacheFromPointer(sharedCache, romClass);`

Reply:
Assuming #7379 gets merged first, I changed all the calls to `offsetInSharedCacheFromPointer` as proposed.